agent/connect/ca/testing.go

```diff
@@ -76,3 +82,155 @@ func TestConsulProvider(t testing.T, d ConsulProviderStateDelegate) *ConsulProvi
 	provider.SetLogger(logger)
 	return provider
 }
+
+// SkipIfVaultNotPresent skips the test if the vault binary is not in PATH.
+//
+// These tests may be skipped in CI. They are run as part of a separate
+// integration test suite.
+func SkipIfVaultNotPresent(t testing.T) {
+	vaultBinaryName := os.Getenv("VAULT_BINARY_NAME")
+	if vaultBinaryName == "" {
+		vaultBinaryName = "vault"
+	}
+
+	path, err := exec.LookPath(vaultBinaryName)
+	if err != nil || path == "" {
+		t.Skipf("%q not found on $PATH - download and install to run this test", vaultBinaryName)
+	}
+}
+
+func NewTestVaultServer(t testing.T) *TestVaultServer {
```

Comment:
I think these kind of helpers are great, but can sometimes make the package harder to use when they are in the `godoc` together with the production stuff.  I generally try to put this type of thing into an `internal/test/vaultca` type package.

But not a big issue, and easy to do as a follow up.